src/librustdoc/doctest/make.rs

```diff
@@ -299,15 +317,32 @@ fn parse_source(
     let dcx = DiagCtxt::new(Box::new(emitter)).disable_warnings();
     let psess = ParseSess::with_dcx(dcx, sm);
 
-    let mut parser = match new_parser_from_source_str(&psess, filename, source) {
+    let mut parser = match new_parser_from_source_str(&psess, filename, wrapped_source) {
         Ok(p) => p,
         Err(errs) => {
             errs.into_iter().for_each(|err| err.cancel());
             cancel_error_count(&psess);
-            return ParsingResult::Failed;
+            return Err(());
         }
     };
-    let mut parsing_result = ParsingResult::Ok;
+
+    fn push_to_s(
+        s: &mut String,
+        source: &str,
```

Comment:
Wait, if you just passed `wrapped_source` we wouldn't need to fix the any of the byte offsets, right? Could we do that instead please

Reply:
Ah, missed that! Fair fair